### PR TITLE
_plot2d: Rotate x-axis ticks if datetime subtype

### PR DESCRIPTION
**Pull request by dcherian**
Rotate x-axis dateticks by default, just as for plot.line()

 - [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Passes ``git diff upstream/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

File: xarray/plot/plot.py [hunk 1]
```diff
@@ -530,6 +530,10 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
 
         _update_axes_limits(ax, xincrease, yincrease)
 
+        # Rotate dates on xlabels
+        if np.issubdtype(xval.dtype, np.datetime64):
+            plt.gcf().autofmt_xdate()
```
Comment (rabernat) [inline xarray/plot/plot.py]: Since the axis object is already known, better to use `ax.autofmt_xdate` than `plt.gcf()...`.

Comment (dcherian) [reply xarray/plot/plot.py]: `autofmt_xdate` is a `Figure` method so that won't work. I could set the rotation manually though...

Comment (rabernat) [reply xarray/plot/plot.py]: ah right, my mistake.

You should be able to use [`ax.get_figure()`](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.get_figure.html#matplotlib.axes.Axes.get_figure) to get the figure. For some reason, `gcf()` seems error prone, but I don't really have any evidence for that.

Comment (dcherian) [reply xarray/plot/plot.py]: Sounds good to me. Changed the corresponding call in `plot.line()` too

File: xarray/tests/test_plot.py [hunk 2]
```diff
@@ -615,6 +615,14 @@ def test_xyincrease_true_changes_axes(self):
         diffs = xlim[0] - 0, xlim[1] - 14, ylim[0] - 0, ylim[1] - 9
         self.assertTrue(all(abs(x) < 1 for x in diffs))
 
+    def test_x_ticks_are_rotated_for_time(self):
+        time = pd.date_range('2000-01-01', '2000-01-10')
+        a = DataArray(np.random.randn(2, len(time)),
+                      [('xx', [1, 2]), ('t', time)])
+        a.plot(x='t')
+        rotation = plt.gca().get_xticklabels()[0].get_rotation()
+        self.assertFalse(rotation == 0)
```
Comment (shoyer) [inline xarray/tests/test_plot.py]: Nit: prefer `self.assertEqual(rotation, 0)`

Comment (dcherian) [reply xarray/tests/test_plot.py]: Yup. Made the change.

**Comment by fmaussion** [comment]
Oh I just noticed I'm able to do it myself. Will wait for the tests to pass and merge

**Pull request merged**